conans/test/build_helpers/msbuild_test.py

```diff
@@ -150,3 +155,114 @@ def build(self):
+                                 "compiler.version": "15",
+                                 "arch": "x86_64",
+                                 "compiler.runtime": "MDd"})
+        conanfile = MockConanfile(settings)
+        msbuild = MSBuild(conanfile)
+        command = msbuild.get_command("dummy.sln", output_binary_log=False)
+        self.assertNotIn("/bl", command)
+
+    def binary_logging_off_implicit_test(self):
+        settings = MockSettings({"build_type": "Debug",
+                                 "compiler": "Visual Studio",
+                                 "compiler.version": "15",
+                                 "arch": "x86_64",
+                                 "compiler.runtime": "MDd"})
+        conanfile = MockConanfile(settings)
+        msbuild = MSBuild(conanfile)
+        command = msbuild.get_command("dummy.sln")
+        self.assertNotIn("/bl", command)
+
+    @unittest.skipUnless(platform.system() == "Windows", "Requires MSBuild")
+    @mock.patch("conans.client.build.msbuild.MSBuild.get_version")
+    def binary_logging_not_supported_test(self, mock_get_version):
+        mock_get_version.return_value = Version("14")
+
+        mocked_settings = MockSettings({"build_type": "Debug",
+                                        "compiler": "Visual Studio",
+                                        "compiler.version": "15",
+                                        "arch": "x86_64",
+                                        "compiler.runtime": "MDd"})
+        conanfile = MockConanfile(mocked_settings)
+        except_text = "MSBuild version detected (14) does not support 'output_binary_log' ('/bl')"
+        msbuild = MSBuild(conanfile)
+
+        with self.assertRaises(ConanException) as exc:
+            msbuild.get_command("dummy.sln", output_binary_log=True)
+        self.assertIn(except_text, str(exc.exception))
+
+    @unittest.skipUnless(platform.system() == "Windows", "Requires MSBuild")
+    def get_version_test(self):
+        settings = MockSettings({"build_type": "Debug",
+                                 "compiler": "Visual Studio",
+                                 "compiler.version": "15",
+                                 "arch": "x86_64",
+                                 "compiler.runtime": "MDd"})
+        version = MSBuild.get_version(settings)
+        self.assertRegexpMatches(version, "(\d+\.){2,3}\d+")
+        self.assertGreater(version, "15.1")
+
+    @unittest.skipUnless(platform.system() == "Windows", "Requires MSBuild")
+    @parameterized.expand([("True",), ("my_log.binlog",)])
+    def binary_log_build_test(self, value):
+        conan_build_vs = """
+from conans import ConanFile, MSBuild
+
+class HelloConan(ConanFile):
+    name = "Hello"
+    version = "1.2.1"
+    exports = "*"
+    settings = "os", "build_type", "arch", "compiler", "cppstd"
+
+    def build(self):
+        msbuild = MSBuild(self)
+        msbuild.build("MyProject.sln", output_binary_log="%s")
```

Comment:
@jgsogo Setting this to ``"%s"`` and formatting with value ``True`` will result in ``output_binary_log="True"`` which is not the desired result and now the test fails...

Reply:
Oh! Sorry, this is a little bit embarrasing :sweat: I'll fix it myself.